autoprotocol/instruction.py

```diff
@@ -86,6 +100,119 @@ def filter_criteria(item):
             ]
         return data
 
+    def _set_informatics(self):
+        """
+        set Instruction informatics to valid list of Informatics
+
+        Raises
+        ------
+        TypeError
+            informatics must be a list
+        TypeError
+            informatics element must be a dict
+        KeyError
+            informatics must have keys 'type' and 'data'
+        ValueError
+            informatics type must be a valid type
+        """
+        # any informatics dict must only have these keys.
+        valid_informatics_keys = ["type", "data"]
+        # informatics must be a list of dict.
+        informatics = self.informatics
+        if isinstance(informatics, list):
+            for info in informatics:
+                if not isinstance(info, dict):
+                    raise TypeError(f"informatics: {info} must be provided in a dict.")
+        else:
+            raise TypeError(
+                f"informatics: {informatics} must be provided in a list of dict. "
+            )
+
+        # instantiate Informatics based on the type
+        valid_informatics = []
+        for info in self.informatics:
+            info_keys = info.keys()
+            if set(info_keys) != set(valid_informatics_keys):
+                raise KeyError(
+                    f"informatics dict: {info} must have keys: {valid_informatics_keys}"
+                )
+            # validate wells only if `wells` param is present in informatics
+            try:
+                if info["data"]["wells"]:
+                    self._check_info_wells()
```

Comment:
nit: this is a generalization that all future informatics instructions with wells will specify that in the `["data"]["wells"]` key. It is a fair assumption to make, but is implicit.
If we want to be stricter, we can instead scope this under the `attach_compounds` check, i.e. 
```
info_obj = AttachCompounds(info["data"])
self._check_info_wells()
```